contracts/ronin/random-beacon/RoninRandomBeacon.sol

```diff
@@ -0,0 +1,407 @@
+          if (_isSubmitted($beacon, id)) {
+            if (unavailableCount != 0) delete _unavailableCount[id];
+          } else {
+            // Increment the consecutive unavailable count and check if it exceeds the threshold
+            _unavailableCount[id] = ++unavailableCount;
+
+            if (unavailableCount >= slashThreshold) {
+              bool slashed = slashIndicator.slashRandomBeacon(id, lastUpdatedPeriod);
+              // Delete the count if the validator has been slashed
+              if (slashed) delete _unavailableCount[id];
+            }
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * @dev Sets the thresholds for multiple validator types.
+   * @notice Emits a `ValidatorThresholdUpdated` event .
+   */
+  function _bulkSetValidatorThresholds(ValidatorType[] calldata validatorTypes, uint256[] calldata thresholds) internal {
+    uint256 length = validatorTypes.length;
+    if (length != thresholds.length) revert ErrLengthMismatch(msg.sig);
+
+    uint256 threshold;
+    ValidatorType validatorType;
+
+    for (uint256 i; i < length; ++i) {
+      validatorType = validatorTypes[i];
+      threshold = thresholds[i];
+
+      _validatorThreshold[validatorType] = threshold;
+
+      emit ValidatorThresholdUpdated(validatorType, threshold);
+    }
+  }
+
+  /**
+   * @dev Filters out the newly joined validators based on the provided threshold.
+   * @param validator The address of the validator.
+   * @param currentPeriod The current period.
+   * @return validCids An array of valid candidate IDs.
+   */
+  function _filterOutNewlyJoinedValidators(
+    address validator,
+    uint256 currentPeriod
+  ) internal view returns (address[] memory validCids) {
+    unchecked {
+      uint256 count;
+      uint256 threshold = COOLDOWN_PERIOD_THRESHOLD;
+      address[] memory allCids = ICandidateManager(validator).getValidatorCandidateIds();
+      uint256[] memory registeredAts = IProfile(getContract(ContractType.PROFILE)).getManyId2RegisteredAt(allCids);
+      uint256 length = allCids.length;
+      validCids = new address[](length);
+
+      for (uint256 i; i < length; ++i) {
+        if (currentPeriod - _toPeriod(registeredAts[i]) >= threshold) {
+          validCids[count++] = allCids[i];
+        }
+      }
+
+      assembly {
+        mstore(validCids, count)
+      }
+    }
+  }
+
+  /**
+   * @dev Checks if a given validator has submitted the random at given period.
+   * @param $beacon The storage reference to the Beacon struct.
+   * @param cid The candidate id to check.
+   * @return true if the validator has submitted the random seed, otherwise false.
+   */
+  function _isSubmitted(Beacon storage $beacon, address cid) internal view returns (bool) {
+    return $beacon.submitted[cid];
+  }
+
+  /**
+   * @dev See {TimingStorage-_computePeriod}.
+   */
+  function _toPeriod(uint256 timestamp) internal view returns (uint256) {
+    return timestamp / PERIOD_DURATION;
```

Comment:
docstring says "computePeriod" but the actual method is transformed to `_toPeriod` somehow? Please be consistent!

Reply:
If this method is a clone  `_computePeriod`, but do it on purpose instead of calling to the `ValidatorSet` contract to reduce the number of internal calls, please comment it out.